test/integration/docker_exec_websocket_server_test.js

```diff
@@ -0,0 +1,92 @@
+var DockerWorker = require('../dockerworker');
+var TestWorker = require('../testworker');
+var assert = require('assert');
+var cmd = require('./helper/cmd');
+var debug = require('debug')('docker-worker:test:docker_exec_websocket_server_test')
+// var waitForEvent = require('../../lib/wait_for_event');
+var slugid = require('slugid');
+var DockerExecClient = require('docker-exec-websocket-server').DockerExecClient;
+var request = require('superagent-promise');
+var Promise = require('promise');
+
+suite('use docker exec websocket server', () => {
+  let worker;
+  setup(async () => {
+    worker = new TestWorker(DockerWorker);
+    await worker.launch();
+  });
+
+  teardown(async () => {
+    if (worker) {
+      await worker.terminate();
+      worker = null;
+    }
+  });
+  test('cat', async () => {
+    let taskId = slugid.v4();
+  	let task = {
+      payload: {
+        image: 'taskcluster/test-ubuntu',
+        command: cmd('sleep 40'),
+        maxRunTime: 2 * 60,
+        features: {
+          interactive: true
+        }
+      }
+    };
+    debug('posting to queue');
+
+    var client;
+
+    let resultPromise = worker.postToQueue(task, taskId);
+    var passed = false;
+    setTimeout(async () => {
+      var getWith303Redirect = async (url) => {
+        var res;
+        try {
+          res = await request.get(url).redirects(0).end();
+        }
+        catch(err) {
+          res = err.response;
+        }
+        return res.headers.location;
+      };
+
+      var url = await getWith303Redirect(worker.queue.buildSignedUrl(
+        worker.queue.getLatestArtifact,
+        taskId,
+        'interactive',
+        {expiration: 60 * 5}));
+
+      assert(url, 'artifact not found');
+
+      client = new DockerExecClient({
+        tty: false,
+        command: ['cat'],
+        url: url,
+      });
+      await client.execute();
+
+      var buf = new Buffer([0xfa, 0xff, 0x0a]);
```

Comment:
I'm curious, why not test with something like: `crypto.randomBytes`
See: https://nodejs.org/api/crypto.html#crypto_crypto_randombytes_size_callback

If we generate say 2mb or 20mb of random data that would prove that it can handle binary data.

Oh, actually it doesn't prove that it's interpreted correctly inside the container... Just that it comes in and out as the same... Which IMO is super close, but maybe it would be interesting to run `md5sum` inside the container and then compare the checksum written out. But that's probably a different test case.